src/main/java/com/github/checkstyle/regression/configuration/ConfigGenerator.java

```diff
@@ -0,0 +1,207 @@
+    private static final String DOCTYPE_SYSTEM =
+            "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd";
+
+    /** The parent name "Checker". */
+    private static final String PARENT_CHECKER = "Checker";
+
+    /** The parent name "TreeWalker". */
+    private static final String PARENT_TREE_WALKER = "TreeWalker";
+
+    /** The name of a module element. */
+    private static final String ELEMENT_MODULE = "module";
+
+    /** The attribute name "name". */
+    private static final String ATTR_NAME = "name";
+
+    /** Prevents instantiation. */
+    private ConfigGenerator() {
+    }
+
+    /**
+     * Creates a new {@link Document} instance from the {@code BASE_CONFIG}.
+     * @return a new {@link Document} instance
+     */
+    private static Document createXmlDocument() {
+        final Document document;
+
+        try {
+            final File baseConfig = new File("src/main/resources/com/github/checkstyle/regression/"
+                    + "configuration/base_config.xml");
+            document = DocumentBuilderFactory.newInstance().newDocumentBuilder()
+                    .parse(baseConfig);
+        }
+        catch (ParserConfigurationException | SAXException | IOException ex) {
+            throw new IllegalStateException("cannot instantiate Document instance", ex);
+        }
+
+        return document;
+    }
+
+    /**
+     * Creates a new {@link Transformer} instance and do initialization.
+     * @return a new {@link Transformer} instance
+     */
+    private static Transformer createXmlTransformer() {
+        final Transformer transformer;
+
+        try {
+            transformer = TransformerFactory.newInstance().newTransformer();
+            transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
+            transformer.setOutputProperty(OutputKeys.DOCTYPE_PUBLIC, DOCTYPE_PUBLIC);
+            transformer.setOutputProperty(OutputKeys.DOCTYPE_SYSTEM, DOCTYPE_SYSTEM);
+        }
+        catch (TransformerConfigurationException ex) {
+            throw new IllegalStateException("cannot instantiate Transformer instance", ex);
+        }
+
+        return transformer;
+    }
+
+    /**
+     * Generates the config XML file from the given module inofs.
+     * @param path        the path of the generated config
+     * @param moduleInfos the given module infos
+     * @return the generated config file
+     * @throws IOException          failure of creating and writing the file
+     * @throws TransformerException failure of transforming the XML document
+     */
+    public static File generateConfig(String path, List<ModuleInfo> moduleInfos)
+            throws IOException, TransformerException {
+        final File file = new File(path);
+        Files.write(file.toPath(), generateConfigText(moduleInfos)
+                .getBytes(Charset.forName("UTF-8")), StandardOpenOption.CREATE_NEW);
```

Comment:
Shouldn't we be using `TRUNCATE_EXISTING` instead of `CREATE_NEW`. We were told a path to place config, we should delete it if it already exists and not fail causing user to erase it manually.